iac/terraform/aws/api_gw.tf

```diff
@@ -34,22 +34,49 @@ resource "aws_api_gateway_deployment" "api_deployment" {
   }
 }
 
-resource "aws_iam_role" "api_gw_cloudwatch_role" {
-  name               = "${var.environment}-api-gw-cloudwatch-role"
+resource "aws_iam_role" "api_gw_role" {
+  name               = "${var.environment}-api-gateway-role"
   assume_role_policy = data.aws_iam_policy_document.api_gw_policy.json
 
   lifecycle {
     create_before_destroy = true
   }
 }
 
+resource "aws_iam_policy" "api_gw_dynamodb_policy" {
+  name        = "${var.environment}-api-gw-dynamodb-policy"
+  description = "Policy allowing API Gateway to query DynamoDB tables"
+
+  policy = jsonencode({
```

Comment:
Do not use jsonencode. Use this
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document